lib/galaxy/datatypes/qualityscore.py

```diff
@@ -41,8 +41,7 @@ def sniff_prefix(self, file_prefix):
         fh = file_prefix.string_io()
         readlen = None
         goodblock = 0
-        while True:
-            line = fh.readline()
+        for line in fh.readlines():
             if not line:
                 if goodblock > 0:
                     return True
```

Comment:
This check is not going to work because the for loop will exit when reaching the EOF, so this need to be moved after the for loop. Also the `break` statements in the loop need to become `return False`.